### PR TITLE
TE-2565 Improve requirements file management

### PR DESCRIPTION
**Pull request by jmbowman**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

Switched over all the requirements files to be generated by pip-tools, but otherwise kept them at their current locations.  This turned up quite a few indirect dependencies we hadn't been pinning before.  Two changes that may merit closer inspection:

* I upgraded awscli in two of the files from 1.15.10 to 1.15.14 to be compatible with the version of boto3 we had chosen.
* I changed how the `aws` role installs Python dependencies to avoid the need to manually update them in two different files.

I'll kick off a from-scratch sandbox build to make sure that works; is there any additional testing that should be done before merging this?

### CODE REVIEW

**Comment by jmbowman** [comment]
The sandbox from scratch got as far as xqueue before failing with the same error as int-nightly, triggered by yesterday's xqueue changes.  I can rebase and try again once that's fixed.

**Review by michaelyoungstrom — APPROVED** [review]
This looks good to me

**Comment by jmbowman** [comment]
Ok, sandbox builds from scratch are now passing: https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/10664/

@feanil , is there anything else I should test before merging this?  I should be mostly online for the next few days at the PyCon development sprints, so I'll be around in case any of the Jenkins jobs or such affected by this have issues with the change.

File: playbooks/roles/aws/templates/requirements.txt.j2 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jmespath==0.9.3           # via botocore
+pyasn1==0.4.2             # via rsa
+python-dateutil==2.7.3    # via botocore, s3cmd
+python-magic==0.4.15      # via s3cmd
+pyyaml==3.12              # via awscli
+rsa==3.4.2                # via awscli
+s3cmd==1.6.1
+s3transfer==0.1.13        # via awscli
+six==1.11.0               # via python-dateutil
+https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
```
Comment (jibsheet) [inline playbooks/roles/aws/templates/requirements.txt.j2]: Our use of cloudformation predates me - I wonder if we can remove it (I'll see if Fred knows, it's really him/Feanil/Ed who might know)

File: requirements/aws.in [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Python dependencies for the aws role
+
+awscli==1.11.58
+boto==2.48.0
+s3cmd==1.6.1
+
+# pip-compile can't deal with this, it's added back by util/post-pip-compile.sh
+# https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
```
Comment (jibsheet) [inline requirements/aws.in]: Yeah, let's just clean this out, I see no reason for you to need to specialcase these in pip-tools.  Let's just remove it.

**Review by jibsheet — APPROVED** [review]
I think you can clean out cloudformation.
I suspect something weird in one of our background jobs might break in an unexpected way.  Would you mind merging on Monday?

**Comment by jmbowman** [comment]
@feanil It looks like dependabot is configured to make PRs directly on what will now be the generated requirements files.  Should we go ahead and reconfigure that or try replacing it with requires.io now?

**Pull request merged**